lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2557,7 +2560,7 @@ HSADispatch::waitComplete() {
         printf("Signal wait returned unexpected value\n");
         exit(0);
     }
-
+    dispose();
```

Comment:
call dispose() here would render the "if (kernargMemory != nullptr)" block below redundant. could you remove it?